File: src/com/darin/weex/WeexApplication.java
package com.darin.weex;

import com.darin.weex.datas.WeexPrimitiveElements;
import com.darin.weex.datas.WeexTemplateCustomEle;
import com.intellij.openapi.components.ApplicationComponent;
import org.jetbrains.annotations.NotNull;

/**
 * Created by DongYayun on 2016/5/14.
 */
public class WeexApplication implements ApplicationComponent {
    public WeexApplication() {
    }

    @Override
    public void initComponent() {
        WeexPrimitiveElements.initPrivitiveEles();
        WeexTemplateCustomEle.initTemplateString();
    }

    @Override
    public void disposeComponent() {
    }

    @Override
    @NotNull
    public String getComponentName() {
        return "WeexApplication";
    }

}

File: src/com/darin/weex/WeexApplication.kt
package com.darin.weex

import com.darin.weex.datas.WeexPrimitiveElements
import com.darin.weex.datas.WeexTemplateCustomEle
import com.intellij.openapi.components.ApplicationComponent

/**
 * Created by DongYayun on 2016/5/14.
 */
class WeexApplication : ApplicationComponent {

    override fun initComponent() {
        WeexPrimitiveElements.initPrivitiveEles()
        WeexTemplateCustomEle.initTemplateString()
    }

    override fun disposeComponent() {
    }

    override fun getComponentName(): String {
        return "WeexApplication"
    }

}
